### DISCUSSION TITLE
Building UI

### DISCUSSION THREAD

**Discussion by v0nerd** [vanna-ai/vanna#197] [Q&A]
This is really stunning! Just it's a pity that we can see the results only on the Jupyter notebook. 
Can we build a lightweight clean frontend which consumes those apis provided by the current backend?

**Comment by zainhoda** [comment]
Hello!

We actually have UIs that you can run:
Streamlit: https://github.com/vanna-ai/vanna-streamlit
Flask: https://github.com/vanna-ai/vanna-flask
Slack: https://github.com/vanna-ai/vanna-slack

**Comment by zainhoda** [comment]
There's another update to this. Now you can run the Flask app from inside the Vanna package:

```python
!pip install vanna
import vanna
from vanna.remote import VannaDefault
vn = VannaDefault(model='chinook', api_key=vanna.get_api_key('my-email@example.com'))
vn.connect_to_sqlite('https://vanna.ai/Chinook.sqlite')
vn.ask('What are the top 10 artists by sales?')

from vanna.flask import VannaFlaskApp
VannaFlaskApp(vn).run()
```

This is launchable from inside a Jupyter notebook or independently

**Comment by qq293580911** [comment]
> There's another update to this. Now you can run the Flask app from inside the Vanna package:
> 
> ```python
> !pip install vanna
> import vanna
> from vanna.remote import VannaDefault
> vn = VannaDefault(model='chinook', api_key=vanna.get_api_key('my-email@example.com'))
> vn.connect_to_sqlite('https://vanna.ai/Chinook.sqlite')
> vn.ask('What are the top 10 artists by sales?')
> 
> from vanna.flask import VannaFlaskApp
> VannaFlaskApp(vn).run()
> ```
> 
> This is launchable from inside a Jupyter notebook or independently

Hello, I am a beginner in Python. I did not find the 'flask' and 'remote' submodules in your source code under src/vanna, which caused my console to continuously report the error: 'ModuleNotFoundError: No module named 'flask''. I hope for an explanation, and I greatly appreciate your help.

**Comment by zainhoda** [comment]
> Hello, I am a beginner in Python. I did not find the 'flask' and 'remote' submodules in your source code under src/vanna, which caused my console to continuously report the error: 'ModuleNotFoundError: No module named 'flask''. I hope for an explanation, and I greatly appreciate your help.

This looks like you don't have the Python package installed. Please join our Discord for more help on this.

**Comment by Alfred212121** [comment]
> There's another update to this. Now you can run the Flask app from inside the Vanna package:现在有另一个更新。现在你可以从Vanna包内部运行Flask应用程序。
> 
> ```python
> !pip install vanna
> import vanna
> from vanna.remote import VannaDefault
> vn = VannaDefault(model='chinook', api_key=vanna.get_api_key('my-email@example.com'))
> vn.connect_to_sqlite('https://vanna.ai/Chinook.sqlite')
> vn.ask('What are the top 10 artists by sales?')
> 
> from vanna.flask import VannaFlaskApp
> VannaFlaskApp(vn).run()
> ```
> 
> This is launchable from inside a Jupyter notebook or independently这可以从Jupyter笔记本内部启动，也可以独立启动

How should I perceive the relationship between the vanna-flask repository and the built-in flask application in vanna? Which one will be prioritized for maintenance in the future?

**Comment by zainhoda** [comment]
> How should I perceive the relationship between the vanna-flask repository and the built-in flask application in vanna? Which one will be prioritized for maintenance in the future?

Thanks for this question. The `vanna-flask` repository as well as all the other UI code (`vanna-streamlit`, `vanna-streamlit-simple`, `vanna-chainlit`, `vanna-slack`, etc) will always be intended as "starting point" for the purposes of customization.

This repo that we're in (the `vanna-ai/vanna` repo) is the main one and the one where future development efforts will be concentrated, particularly with future community contributions.
